ssl/quic/quic_obj_local.h

```diff
@@ -0,0 +1,280 @@
+     * this is the root object.
+     */
+    QUIC_OBJ                *parent_obj;
+
+    /* invariant: != NULL */
+    QUIC_OBJ                *cached_event_leader;
+    /* invariant: != NULL iff this is a port leader or subsidiary object */
+    QUIC_OBJ                *cached_port_leader;
+
+    /*
+     * Points to the QUIC_ENGINE instance. Always equals
+     * cached_event_leader->engine. The containing_obj APL object owns this
+     * instance iff is_event_leader is set, otherwise it is an additional
+     * reference cached for convenience. Unlike port this is never NULL because
+     * a QUIC domain is always rooted in an event leader.
+     */
+    QUIC_ENGINE             *engine;
+
+    /*
+     * Points to the QUIC_PORT instance applicable to the containing_obj APL
+     * object, or NULL if we are not at or below a port leader. Always equals
+     * cached_port_leader->port. The containing_obj APL object owns this
+     * instance iff is_port_leader is set, otherwise it is an additional
+     * reference cached for convenience.
+     */
+    QUIC_PORT               *port;
+
+    unsigned int            init_done       : 1;
+    unsigned int            is_event_leader : 1;
+    unsigned int            is_port_leader  : 1;
+};
+
+/*
+ * Initialises a QUIC_OBJ structure with zero or more roles active. Returns 1
+ * on success or 0 on failure.
+ *
+ * ctx: A SSL_CTX used to initialise the SSL base object structure.
+ *
+ * type: A SSL_TYPE_* value designating the SSL object type.
+ *
+ * parent_obj: NULL if this is the root APL object in a new hierarchy, or a
+ * pointer to the parent APL object otherwise.
+ *
+ * engine: If non-NULL, this object becomes the Event Leader. parent_obj must be
+ * NULL iff this is non-NULL as currently the Event Leader is always the root in
+ * an APL object hierarchy. If NULL, the contextually applicable engine is
+ * determined by using parent_obj and ancestors to find the Event Leader.
+ *
+ * port: If non-NULL, this object becomes a Port Leader. If NULL, the
+ * contextually applicable port (if any) is determined by using parent_obj and
+ * ancestors to find the Port Leader.
+ */
+int ossl_quic_obj_init(QUIC_OBJ *obj,
+                       SSL_CTX *ctx,
+                       int type,
+                       SSL *parent_obj,
+                       QUIC_ENGINE *engine,
+                       QUIC_PORT *port);
+
+/*
+ * Returns a pointer to the handshake layer object which should be accessible on
+ * obj for purposes of handshake API autoforwarding, if any.
+ *
+ * This returns NULL if a handshake layer SSL object is available but should not
+ * be used for autoforwarding purposes, for example on a QSSO.
+ */
+SSL_CONNECTION *ossl_quic_obj_get0_handshake_layer(QUIC_OBJ *obj);
+
+/*
+ * Returns a pointer to the SSL base object structure. Returns NULL if obj is
+ * NULL. If obj is non-NULL, it must be initialised.
+ */
+static ossl_inline ossl_unused SSL *
+ossl_quic_obj_get0_ssl(QUIC_OBJ *obj)
```

Comment:
It is necessary to break many of these over multiple lines to keep the lines within length... Or is your comment related to where a `*` is placed?

Reply:
in [c-style](https://www.openssl.org/policies/technical/coding-style.html) we often show function definition like this
```
int function(int x)
{
        body of function
}
```
So I've just jumped to wrong assumption. In fact what you have currently is style I prefer more. Because I can quickly find function definition by just using search pattern `/^func_name`.

the c-style guide does not talk at all on how functions should be defined.